pynest/examples/hpc_benchmark.py

```diff
@@ -0,0 +1,426 @@
+                 {'rule': 'fixed_indegree', 'indegree': CE, 'autapses': False, 'multapses': True},
+                 {'model': 'stdp_pl_synapse_hom_hpc'})
+
+    nest.message(M_INFO, 'build_network', 'Connecting inhibitory -> excitatory population.')
+
+    nest.Connect(I_neurons, E_neurons,
+                 {'rule': 'fixed_indegree', 'indegree': CI, 'autapses': False, 'multapses': True},
+                 {'model': 'static_synapse_hpc', 'weight': JE_pA * brunel_params['g']})
+
+    nest.message(M_INFO, 'build_network', 'Connecting excitatory -> inhibitory population.')
+
+    nest.Connect(E_neurons, I_neurons,
+                 {'rule': 'fixed_indegree', 'indegree': CE, 'autapses': False, 'multapses': True},
+                 {'model': 'static_synapse_hpc', 'weight': JE_pA})
+
+    nest.message(M_INFO, 'build_network', 'Connecting inhibitory -> inhibitory population.')
+
+    nest.Connect(I_neurons, I_neurons,
+                 {'rule': 'fixed_indegree', 'indegree': CI, 'autapses': False, 'multapses': True},
+                 {'model': 'static_synapse_hpc', 'weight': JE_pA * brunel_params['g']})
+
+    if params['record_spikes']:
+        local_neurons = list(get_local_nodes(E_neurons))
+
+        if len(local_neurons) < brunel_params['Nrec']:
+            nest.message(M_ERROR, 'build_network',
+                         '''Spikes can only be recorded from local neurons, but the number of local
+                         neurons is smaller than the number of neurons spikes should be recorded from.
+                         Aborting the simulation!''')
+            exit(1)
+
+        nest.message(M_INFO, 'build_network', 'Connecting spike detectors.')
+        nest.Connect(local_neurons[:brunel_params['Nrec']], E_detector)
+
+    # read out time used for building
+    BuildEdgeTime = time.time() - tic
+
+    logger.log(str(BuildEdgeTime) + ' # build_edge_time')
+    logger.log(str(memory_thisjob()) + ' # virt_mem_after_edges')
+
+
+def run_simulation():
+    # open log file
+    logger = Logger(params['log_file'])
+
+    nest.ResetKernel
+    nest.set_verbosity(M_INFO)
+
+    logger.log(str(memory_thisjob()) + ' # virt_mem_0')
+
+    build_network(logger)
+
+    tic = time.time()
+
+    nest.Simulate(params['presimtime'])
+
+    PreparationTime = time.time() - tic
+
+    logger.log(str(memory_thisjob()) + ' # virt_mem_after_presim')
+    logger.log(str(PreparationTime) + ' # presim_time')
+
+    tic = time.time()
+
+    nest.Simulate(params['simtime'])
+
+    SimCPUTime = time.time() - tic
+
+    logger.log(str(memory_thisjob()) + ' # virt_mem_after_sim')
+    logger.log(str(SimCPUTime) + ' # sim_time')
+
+    logger.log(str(compute_rate()) + ' # average rate')
+
+    logger.done()
+
+# ------------------------------------------------------------------------------------
+
+
+def compute_rate():
+    '''compute approximation of average firing rate based on number of
+    local nodes, number of local spikes and total time
+    '''
```

Comment:
There should be one common format for docstrings of functions across the script.
Either like this:
'''...
'''

or 
'''
...
'''